warehouse/sponsors/tasks.py

```diff
@@ -0,0 +1,70 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from urllib.parse import urlencode
+
+import requests
+
+from sqlalchemy import or_, true
+from sqlalchemy.exc import NoResultFound
+
+from warehouse import tasks
+from warehouse.sponsors.models import Sponsor
+
+
+@tasks.task(ignore_result=True, acks_late=True)
+def update_pypi_sponsors(request):
+    """
+    Read data from pythondotorg's logo placement API and update Sponsors
+    table to mirror it.
+    """
+    host = request.registry.settings["pythondotorg.host"]
+    token = request.registry.settings["pythondotorg.api_token"]
+    headers = {"Authorization": f"Token {token}"}
+
```

Comment:
As an alternate to https://github.com/pypa/warehouse/pull/10766/files#r808505811 we could short circuit here if host and token are not both set.

Reply:
Or maybe we could just prevent the task from being registered at all?

Reply:
yeah, that comment is what's linked :-D I'll resolve this one and let that suggestion stand alone